agora-test/Spec/Sample/Proposal.hs

```diff
@@ -0,0 +1,198 @@
+
+--------------------------------------------------------------------------------
+
+governor :: Governor
+governor = Governor
+
+govPolicy :: MintingPolicy
+govPolicy = mkMintingPolicy (governorPolicy governor)
+
+govValidator :: Validator
+govValidator = mkValidator (governorValidator governor)
+
+govSymbol :: CurrencySymbol
+govSymbol = mintingPolicySymbol govPolicy
+
+proposal :: Proposal
+proposal =
+  Proposal
+    { governorSTAssetClass =
+        -- TODO: if we had a governor here
+        AssetClass
+          ( govSymbol
+          , ""
+          )
+    }
+
+-- | 'Proposal' policy instance.
+policy :: MintingPolicy
+policy = mkMintingPolicy (proposalPolicy proposal)
+
+policySymbol :: CurrencySymbol
+policySymbol = mintingPolicySymbol policy
+
+-- | A sample 'PubKeyHash'.
+signer :: PubKeyHash
+signer = "8a30896c4fd5e79843e4ca1bd2cdbaa36f8c0bc3be7401214142019c"
+
+-- | 'Proposal' validator instance.
+validator :: Validator
+validator = mkValidator (proposalValidator proposal)
+
+-- | 'TokenName' that represents the hash of the 'Proposal' validator.
+validatorHashTN :: TokenName
+validatorHashTN = let ValidatorHash vh = validatorHash validator in TokenName vh
+
+propThresholds :: ProposalThresholds
+propThresholds =
+  ProposalThresholds
+    { countVoting = Tagged 1000
+    , create = Tagged 1
+    , vote = Tagged 10
+    }
+
+-- | This script context should be a valid transaction.
+proposalCreation :: ScriptContext
+proposalCreation =
+  let st = Value.singleton policySymbol "" 1 -- Proposal ST
+      proposalDatum :: Datum
+      proposalDatum =
+        Datum
+          ( toBuiltinData $
+              ProposalDatum
+                { proposalId = ProposalId 0
+                , effects =
+                    AssocMap.fromList
+                      [ (ResultTag 0, [])
+                      , (ResultTag 1, [])
+                      ]
+                , status = Draft
+                , cosigners = [signer]
+                , thresholds = propThresholds
+                , votes = ProposalVotes $ AssocMap.empty
+                }
+          )
+
+      govBefore :: Datum
+      govBefore =
+        Datum
+          ( toBuiltinData $
+              GovernorDatum
+                { proposalThresholds = propThresholds
+                , nextProposalId = ProposalId 0
+                }
+          )
+      govAfter :: Datum
+      govAfter =
+        Datum
+          ( toBuiltinData $
+              GovernorDatum
+                { proposalThresholds = propThresholds
+                , nextProposalId = ProposalId 1
+                }
+          )
+   in ScriptContext
+        { scriptContextTxInfo =
+            TxInfo
+              { txInfoInputs =
+                  [ TxInInfo
+                      (TxOutRef "0b2086cbf8b6900f8cb65e012de4516cb66b5cb08a9aaba12a8b88be" 1)
+                      TxOut
+                        { txOutAddress = Address (ScriptCredential $ validatorHash validator) Nothing
```

Comment:
I think it should be `govValidator` here?

Reply:
Yeah, you're right. Though it technically doesn't matter for the Proposal -- all it cares about is that the token moves. It _will_ matter once this is also validated by the governor validator, though.